content/en/docs/usage/make_build.md

```diff
@@ -0,0 +1,352 @@
+* Add new source files to the build.
+* Use different configuration options.
+
+Notice that the `app-helloworld` directory already contains a `Makefile` that looks like this:
+
+```Makefile
+UK_ROOT ?= $(PWD)/../../unikraft
+UK_LIBS ?= $(PWD)/../../libs
+LIBS :=
+
+all:
+	@$(MAKE) -C $(UK_ROOT) A=$(PWD) L=$(LIBS)
+
+$(MAKECMDGOALS):
+	@$(MAKE) -C $(UK_ROOT) A=$(PWD) L=$(LIBS) $(MAKECMDGOALS)
+```
+
+In the first two lines, we provide the paths to the Unikraft core repository and the directory where we can put additional libraries.
+Since the `helloworld` application uses [`nolibc`](https://github.com/unikraft/unikraft/tree/staging/lib/nolibc) as C library, which comes with the Unikraft core, we do not have any external dependencies.
+We thus do not have to download or reference any libraries in the `Makefile` and can leave the `LIBS` variable empty.
+
+When the application becomes more complex, it is likely that not all functionality is included in the Unikraft core repository.
+As you might have noticed already from the code organization in the core repository (`lib/*`), Unikraft puts an emphasis on encapsulating functionality in libraries that can be easily added, removed, or replaced to tailor the unikernel to the application's needs.
+
+To add a library, clone it into the `libs` directory created earlier.
+In this example, we add the `lightweight IP` (`lwip`) network stack [`lib-lwip`](https://github.com/unikraft/lib-lwip):
+
+```console
+$ cd my-unikernel
+$ cd libs
+$ git clone https://github.com/unikraft/lib-lwip
+```
+
+Next, we have reference the library in the `Makefile` of `app-helloworld` so that it becomes available for selection in the configuration:
+
+```Makefile
+UK_LIBS ?= $(PWD)/../../libs
+
+LIBS := $(UK_LIBS)/lib-lwip
+#LIBS := $(LIBS-y):$(UK_LIBS)/lib-X
+```
+
+We can now re-run
+
+```console
+$ make menuconfig
+```
+
+and select `lwip` in the `Library Configuration` sub-menu.
+We can then rebuild the unikernel, by running:
+
+```console
+$ make clean
+$ make prepare
+$ make
+```
+
+The `prepare` step gives the build system the chance to pull the actual source code for `lwip` and apply necessary patches to make it work with Unikraft.
+This has to be done only once after adding a new library.
+
+Note that the build may fail because depending on the configuration, `lwip` requires a more extensive `libc` (e.g. [`musl`](https://github.com/unikraft/lib-musl)).
+You can add these additional dependencies just as described.
+
+To add new source files to the build system, we add them to the `Makefile.uk` file as such:
+
+```makefile
+$(eval $(call addlib,apphelloworld))
+
+APPHELLOWORLD_SRCS-y += $(APPHELLOWORLD_BASE)/main.c
+APPHELLOWORLD_SRCS-y += $(APPHELLOWORLD_BASE)/source-file-2.c
+```
+
+For more detailed information about using and configuring the build system, please refer to the [porting page](docs/develop/porting/).
+
```

Comment:
Add an example where you use `Config.uk` to select `lwip`, `musl`, and also enable logging. This is the third extension, under a triple-hash heading (`###`).